src/src/Helpers/TarStream.h

```diff
@@ -0,0 +1,175 @@
+/**
+ * TarStream : Create/receive a .tar file while streaming via http webserver
+ * Copyright (c) 2023.. Ton Huisman for ESPEasy
+ *
+ * Changelog:
+ * 2023-08-24 tonhuisman: Implement streaming in a .tar storing all regular files in the chosen storage (Flash or SD), replacing existing
+ *                        files, Flash: adding/replacing only if there is at least 2 blocks of storage available
+ * 2023-08-23 tonhuisman: Implement streaming out a .tar via the read() method
+ * 2023-08-19 tonhuisman: Initial setup
+ */
+
+#ifndef HELPERS_TAR_STREAM_H
+#define HELPERS_TAR_STREAM_H
+
+#include "../../ESPEasy_common.h"
+
+#if FEATURE_TARSTREAM_SUPPORT
+
+# include "../Helpers/ESPEasy_Storage.h"
+
+# include <FS.h>
+# include <vector>
+
+// This are internal features only, to be used when debugging the code
+# define TAR_STREAM_DEBUG    0               // Include/exclude some logging
+# define TAR_STREAM_PEEK     0               // Include/exclude peek()
+# define TAR_LOG_LEVEL_DEBUG LOG_LEVEL_DEBUG // Can use DEBUG or INFO
+
+/* tar Header Block, from POSIX 1003.1-1990.  */
+
+// Source: https://www.gnu.org/software/tar/manual/html_node/Standard.html
+
+/* POSIX header.  */
+
+struct posix_header       /* Using POSIX Tar-definitions for accuracy */
```

Comment:
Maybe you should also add a check for the size of this struct and give an assert if it doesn't match the expected size. (compile time checks)

Reply:
Check added